internal/provider/resource_nd_site.go

```diff
@@ -0,0 +1,491 @@
+
+		if siteType == "aci" || siteType == "cloud_aci" {
+			siteTypeParam := siteType
+			if siteType == "cloud_aci" {
+				siteTypeParam = siteTypeMap[siteType]
+			}
+
+			siteConfiguration[siteTypeParam] = map[string]interface{}{
+				"InbandEPGDN": inbandEpg,
+			}
+		} else if siteType == "ndfc" || siteType == "dcnm" {
+			siteConfiguration[siteType] = map[string]string{
+				"fabricName":       payloadMap["name"].(string),
+				"fabricTechnology": "External",
+				"fabricType":       "External",
+			}
+		}
+	}
+
+	payloadMap["siteConfig"] = siteConfiguration
+	payloadMap["siteType"] = siteTypeMap[siteType]
+
+	payload, err := json.Marshal(map[string]interface{}{"spec": payloadMap})
+	if err != nil {
+		diags.AddError(
+			"Marshalling of json payload failed",
+			fmt.Sprintf("Err: %s. Please report this issue to the provider developers.", err),
+		)
+		return nil
+	}
+
+	jsonPayload, err := gabs.ParseJSON(payload)
+
+	if err != nil {
+		diags.AddError(
+			"Construction of json payload failed",
+			fmt.Sprintf("Err: %s. Please report this issue to the provider developers.", err),
+		)
+		return nil
+	}
+	return jsonPayload
+}
+
+func setSiteId(ctx context.Context, data *SiteResourceModel) {
+	data.Id = types.StringValue(data.SiteName.ValueString())
+}
+
+func getAndSetSiteAttributes(ctx context.Context, diags *diag.Diagnostics, client *Client, data *SiteResourceModel) {
+
+	responseData := DoRestRequest(ctx, diags, client, fmt.Sprintf("%s/%s", sitePath, data.Id.ValueString()), "GET", nil)
+	*data = *getBaseSiteResourceModel(data.SiteUsername.ValueString(), data.SitePassword.ValueString(), data.LoginDomain.ValueString())
+
+	if diags.HasError() {
+		return
+	}
+
+	if responseData.Data() != nil {
+		responseReadInfo := responseData.Data().(map[string]interface{})
+		specReadInfo := responseReadInfo["spec"].(map[string]interface{})
+		for attributeName, attributeValue := range specReadInfo {
+			if attributeName == "name" {
+				data.SiteName = basetypes.NewStringValue(attributeValue.(string))
+				data.Id = basetypes.NewStringValue(attributeValue.(string))
+			}
+
+			if attributeName == "siteConfig" {
+				data.InbandEpg = basetypes.NewStringValue(attributeValue.(map[string]interface{})[siteTypeMap[specReadInfo["siteType"].(string)]].(map[string]interface{})["InbandEPGDN"].(string))
+			}
+
+			if attributeName == "host" {
+				data.Url = basetypes.NewStringValue(attributeValue.(string))
+			}
+
+			if attributeName == "siteType" {
+				data.SiteType = basetypes.NewStringValue(siteTypeMap[attributeValue.(string)])
+			}
+
+			if attributeName == "latitude" {
+				data.Latitude = basetypes.NewStringValue(attributeValue.(string))
+			}
+
+			if attributeName == "longitude" {
+				data.Longitude = basetypes.NewStringValue(attributeValue.(string))
+			}
+
+			if os.Getenv("ND_LOGIN_DOMAIN") != "" {
+				data.LoginDomain = basetypes.NewStringValue(os.Getenv("ND_LOGIN_DOMAIN"))
+			} else if attributeName == "loginDomain" {
```

Comment:
This part of the code is not required anymore because data.LoginDomain was already set using getBaseSiteResourceModel
```
else if attributeName == "loginDomain" {
				data.LoginDomain = basetypes.NewStringValue(attributeValue.(string))
			}
```